xarray/backends/api.py

```diff
@@ -451,23 +451,29 @@ def open_dataset(
     decode_cf : bool, optional
         Whether to decode these variables, assuming they were saved according
         to CF conventions.
-    mask_and_scale : bool, optional
+    mask_and_scale : bool or dict, optional
```

Comment:
```suggestion
    mask_and_scale : bool or dict-like, optional
```
I think we use dict-like in similar situations. Although, we are not very consistent there.
And while at it: add this to the other parameters as well!